components/style/rule_tree/mod.rs

```diff
@@ -660,6 +678,12 @@ impl StrongRuleNode {
         }
     }
 
+    /// Returns whether this node has any child, only intended for testing
+    /// purposes.
+    pub unsafe fn has_children_for_testing(&self) -> bool {
+        !self.get().first_child.load(Ordering::Relaxed).is_null()
```

Comment:
This is only for testing, and there are barriers around (this is only executed in the "sequential" mode).